util/zstdframe/options.go

```diff
@@ -144,8 +181,10 @@ func getDecoder(opts ...Option) decoder {
 		switch opt := opt.(type) {
 		case maxDecodedSize:
 			maxSize = uint64(opt)
-			dopts.maxSizeLog2 = 64 - bits.LeadingZeros64(maxSize-1)
+			dopts.maxSizeLog2 = uint8(64 - bits.LeadingZeros64(maxSize-1))
 			dopts.maxSizeLog2 = min(max(10, dopts.maxSizeLog2), 63)
+		case maxWindowSize:
+			dopts.maxWindowLog2 = uint8(64 - bits.LeadingZeros64(uint64(opt)-1))
```

Comment:
The min window log is also 10, and is max of 30 in 32-bit implementations and 31 in 64-bit implementations.

Reply:
The 32-bit limit is a quirk of the current "zstd" package implementation. The RFC permits it to be up to ~4TB.

If the `zstd.MaxWindowSize` constant is changed due to an internal implementation detail, I'd rather not have this package accidentally break.